models/issue_comment.go

```diff
@@ -574,9 +574,13 @@ func (c *Comment) LoadPushCommits() (err error) {
 		defer gitRepo.Close()
 
 		c.Commits = gitRepo.GetCommitsFromIDs(commitIDs)
-		c.Commits = ValidateCommitsWithEmails(c.Commits)
-		c.Commits = ParseCommitsWithSignature(c.Commits, c.Issue.Repo)
-		c.Commits = ParseCommitsWithStatus(c.Commits, c.Issue.Repo)
+		if c.Commits.Len() > 0 {
+			c.Commits = ValidateCommitsWithEmails(c.Commits)
+			c.Commits = ParseCommitsWithSignature(c.Commits, c.Issue.Repo)
+			c.Commits = ParseCommitsWithStatus(c.Commits, c.Issue.Repo)
+		} else {
+			c.Commits = nil
```

Comment:
Isn't it useful to keep `c.Commits` here? So if by some twisted logic this function is called twice, all the checks can be skipped.

I can see it makes the `tmpl` look cleaner, though. 🤔

Reply:
Why it's not usefull? It's used to show message on ui and emal. And to protect it from called twice, I have add ``c.Commits != nil`` check in here. 
https://github.com/a1012112796/gitea/blob/57fcd825b3f37e03a1192ff0487ec9648be46ce5/models/issue_comment.go#L556
maybe I am not fullly understand your thought , please give me more detail . Thanks

Reply:
>
> maybe should also remove c.Content , Then It will not be double called.

No, please don't alter the model as a replacement of flags; that's hacky and difficult to maintain.

What I understood from the code above is this: `IF c.Commits.Len() == 0 THEN c.Commits = nil`, so you're "nilling" c.Commits if it has no elements. I thought to my self: "hey, why not leave c.Commits as an empty list instead of nilling it so the next time this function is called, it's bypassed entirely thanks to the if at the beginning of the function?". 😄